doc/tutorials/introduction/crosscompilation/crosscompile_with_multiarch.markdown

```diff
@@ -0,0 +1,341 @@
+--   Version control:               4.8.0-xxx-XXXXXXXX
+--
+--   Extra modules:
+--     Location (extra):            /home/kmtr/work/opencv_contrib/modules
+--     Version control (extra):     4.8.0-xxx-XXXXXXXX
+--
+--   Platform:
+--     Timestamp:                   yyyy-mm-ddThh:mm:ssZ
+--     Host:                        Linux 6.2.0-32-generic x86_64
+--     CMake:                       3.25.1
+```
+
+Update apt and dpkg settings.
+----------------------------
+apt and dpkg is package management system using ubuntu and debian.
+
+Following are setup steps to use MultiArch.
+
+### Step1)Add apt source
+
+Execute `sudo apt edit-sources` to add foreign arch libraries at end of file.
+
+ex1) arm64 and armv7 for Ubuntu 23.04
+
+```
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-updates main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-updates universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-updates multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-backports main restricted universe multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-security main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-security universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports lunar-security multiverse
+```
+
+ex2) arm64 and armv7 for Ubuntu 23.10
+
+```
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-updates main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-updates universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-updates multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-backports main restricted universe multiverse
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-security main restricted
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-security universe
+deb [arch=arm64,armhf] http://ports.ubuntu.com/ubuntu-ports mantic-security multiverse
+```
+
+### Step2) Update apt database
+Update apt database to apply new apt sources.
+
+```
+[Host]
+apt update
+```
+
+### Step3) Update dpkg setting
+Update dpkg settings to support foreign architectures.
+
+With `--print-architecture` and `--print-foreign-architectures`, we can confirm that dpkg works well.
+
+```
+[Host]
+$ sudo dpkg  --add-architecture arm64
+$ sudo dpkg  --add-architecture armhf
```

Comment:
There is a mix of terminal snippets - starting with `$` and without. Prefer avoid `$` as documentation does not support copy-to-clipboard option.

Reply:
I was worried about how to distinguish between lines input by the user and lines output by the program. So I added a `$` sign to the beginning of each line. 

However, it becomes a problem when copying and pasting. These have been removed.

`[Host]` and `[Target]` to indicate the execution environment also seem to get in the way... I think I'll remove this tomorrow too. Thank you !

Reply:
Yes, I fixed them. In recent document I think it is suitable to copy and paste ! Thank you for your point!

Reply:
Maybe something like this?

### Step3) Update dpkg setting
Update dpkg settings to support foreign architectures.

Execute `dpkg --add-architecture` with a target architecture:

```
sudo dpkg --add-architecture arm64
sudo dpkg --add-architecture armhf
```

And the following comand shows what foreign architectures are supported.

```
sudo dpkg --print-architecture
```
```
arm64
armhf
```

Reply:
Yes, I split bach script and its result, thank you for your suggestion.

(It was also used for `ldd` command to detect dependency section.)